tools/tungsten/core/src/com/google/android/filament/tungsten/model/Nodes.kt

```diff
@@ -140,6 +141,34 @@ private val constantFloat2NodeCompile = fun(node: Node, compiler: GraphCompiler)
     return node
 }
 
+private val textureSampleCompile = fun(node: Node, compiler: GraphCompiler): Node {
+    compiler.requireAttribute("uv0")
+    val parameter = compiler.addParameter("sampler2d", "texture")
+    compiler.associateParameterWithProperty(parameter, node.getPropertyHandle("textureSource"))
+
+    val outputVariable = compiler.getNewTemporaryVariableName("textureSample")
+    compiler.addCodeToMaterialFunctionBody(
+            "float4 $outputVariable = texture(materialParams_${parameter.name}, getUV0());\n", 4)
```

Comment:
We should give the ability to plug a UV set in the texture note, and default to UV0 by default